Tasks/Maven/sonarqube-common/taskreport.ts

```diff
@@ -0,0 +1,42 @@
+import tl = require('vsts-task-lib/task');
+
+// Data class representing the report-task.txt file contents.
+export class TaskReport {
+
+    constructor(public projectKey: string,
```

Comment:
This class should have a single constructor that takes in the location of the the report-task.txt (or better yet, it figures it out on its own) and deserializes it. It should then expose the serverUrl, dashboardUrl etc. as read-only properties.